dissect/target/loaders/cb.py

```diff
@@ -1,143 +1,197 @@
     RegistryValueNotFoundError,
 )
 from dissect.target.filesystems.cb import CbFilesystem
 from dissect.target.helpers.regutil import RegistryHive, RegistryKey, RegistryValue
+from dissect.target.helpers.utils import parse_path_uri
 from dissect.target.loader import Loader
 from dissect.target.plugins.os.windows.registry import RegistryPlugin
 
+if TYPE_CHECKING:
+    from dissect.target.target import Target
+
 
 class CbLoader(Loader):
-    def __init__(self, path, **kwargs):
-        path = str(path)
+    def __init__(self, path: str, parsed_path: ParseResult = None, **kwargs):
+        self.host, _, instance = parsed_path.netloc.partition("@")
         super().__init__(path)
-        self.uri = urlparse(path)
-        self.host = self.uri.netloc
 
-        self.cb = CbResponseAPI()
-        if self.host.isdigit():
-            self.sensor = self.cb.select(Sensor, self.host)
-        else:
-            q = self.cb.select(Sensor)
+        # A profile will need to be given as argument to CBCloudAPI
+        # e.g. cb://workstation@instance
+        try:
+            self.cbc_api = CBCloudAPI(profile=instance)
+        except CredentialError:
+            raise LoaderError("The Carbon Black Cloud API key was not found or has the wrong set of permissions set")
+
+        self.sensor = self.get_device()
+        if not self.sensor:
+            raise LoaderError("The device was not found within the specified instance")
 
-            if all([part.isdigit() for part in self.host.split(".")]):
-                q = q.where(f"ip:{self.host}")
+        self.session = self.sensor.lr_session()
+
+    def get_device(self) -> Optional[Device]:
+        host_is_ip = self.host.count(".") == 3 and all([part.isdigit() for part in self.host.split(".")])
+
+        for cbc_sensor in self.cbc_api.select(Device).all():
+            if host_is_ip:
+                if cbc_sensor.last_internal_ip_address == self.host:
+                    return cbc_sensor
             else:
-                q = q.where(f"hostname:{self.host}")
+                try:
+                    device_name = cbc_sensor.name.lower()
+                except AttributeError:
+                    continue
 
-            res = list(q)
-            if len(res) > 1:
-                res = sorted(res, key=lambda s: s.last_checkin_time, reverse=True)
+                # Sometimes the domain name is included in the device name
+                # E.g. DOMAIN\\Hostname
+                if "\\" in device_name:
+                    device_name = device_name.split("\\")[1]
 
-            self.sensor = res[0]
+                if device_name == self.host.lower():
+                    return cbc_sensor
 
-        self.session = self.sensor.lr_session()
+        return None
 
     @staticmethod
-    def detect(path):
-        return urlparse(str(path)).scheme == "cb"
+    def detect(path: Path) -> bool:
+        path_part, _, _ = parse_path_uri(path)
+        return path_part == "cb"
```

Comment:
```suggestion
        return False
```
Just return False here, URI notation will force this loader anyway.